Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m

```diff
@@ -352,13 +352,14 @@ - (void)configureViews
             if (self.hasBodyMediaWithThumbnail) {
                 [self.stackView addArrangedSubview:bodyMediaView];
 
-                OWSBubbleShapeView *strokeView = [OWSBubbleShapeView bubbleDrawView];
-                strokeView.strokeThickness = CGHairlineWidth();
-                strokeView.strokeColor = (Theme.isDarkThemeEnabled ? [UIColor colorWithWhite:1.f alpha:0.2f]
-                                                                   : [UIColor colorWithWhite:0.f alpha:0.2f]);
-                [bodyMediaView addSubview:strokeView];
-                [self.bubbleView addPartnerView:strokeView];
-                [self.viewConstraints addObjectsFromArray:[strokeView ows_autoPinToSuperviewEdges]];
+                OWSBubbleShapeView *innerShadowView = [OWSBubbleShapeView bubbleInnerShadowView];
+                innerShadowView.innerShadowColor
+                    = (Theme.isDarkThemeEnabled ? UIColor.ows_whiteColor : UIColor.ows_blackColor);
+                innerShadowView.innerShadowRadius = 0.5f;
```

Comment:
I'm not sure what the perf implications for this are, but if we're setting three properties, and each triggers `updateLayers`, maybe we should instead have a single three-arg-setter which does a single `updateLayers`

TIOLI though - maybe this method isn't a hotspot.

Reply:
Replaced the factory methods with initializers that take the required configuration for each mode to avoid this.  I'm not worried about the perf though.